src/cljam/tools/cli.clj

```diff
@@ -68,7 +71,13 @@
                                  "bam"  (sam/bam-reader f))]
         (when (:header options)
           (println (sam-util/stringify-header (sam/read-header r))))
-        (doseq [aln (sam/read-alignments r)]
+        (doseq [aln (if (:region options)
```

Comment:
Why not adding `indexed?` to protocol `IAlignmentReader` or `IReader`?

Reply:
Your question is reasonable. I hesitated over adding it to such protocols because they are also used for non-indexed readers like `SAMReader`. There are the following solutions:

* Not change - `indexed?` is internal function for the present.
* Add it to such protocols. Non-indexed readers always return `false`.
* Add a new protocol `IIndexedReader`.

I want to hear your opinion. Thanks.

Reply:
Considering that we would have `CRAMReader` or DB or some other serialization formats in the future and that `SAMReader` already accepts `read-alignments` with 2 args, I think the 2nd option fits better to keep the code simple.
> Add it to such protocols. Non-indexed readers always return false.

As we want to know if a reader has an index or not,  a protocol to show a format is indexable is not so important.

Reply:
OK, but I just realized that `indexed?` cannot be added to both `IAlignmentReader` and `ISequenceReader` because of name conflict. So it should be added to `IReader` and defined as a function checking the protocol can be random accessed.